server/engine/telemetry.go

```diff
@@ -63,6 +63,8 @@ func PublishToSegment(db *gorm.DB) {
 	model.SetMetric(db, model.ApplicationId, config.GetEnvironment().APPLICATION_ID)
 	// time.RFC3339 format is the Go equivalent to ISO 8601 format (minus the milliseconds)
 	model.SetMetric(db, model.EndTime, time.Now().Format(time.RFC3339))
+	// time.RFC3339 format is the Go equivalent to ISO 8601 format (minus the milliseconds)
```

Comment:
What's the rationale behind adding these timestamps? Shouldn't they come from the callers and we only add them if they are missing?
With this implementation, the will be mostly of the same value, and if that is the intent here, then we will be better off getting and formatting the time once and setting it for both `StartTime` and `EndTime`.

Reply:
The EndTime represents the Installer's end time i.e the time that the Installer finished the deployment and is about to go into its 15 minute hibernation period before shutting off. Since this is specific to the Installer itself, it needs to be added in the Installer itself.

The StartTime is meant to represent the start time of the deployment itself i.e the time at which the managed app deployment commences. Hence, this will require an addition in the main.bicep and containerInstance.bicep and then have that flow down to the Installer where it can be consumed, stored into the telemetry table and published from there.